scalems/wrappers/__init__.py

```diff
@@ -16,32 +15,109 @@
+        mathematically. The chemical system, instructions for running the simulation, but 
+        not how it's executed, or the resources it will take. That will be taken care of later in
+        the workflow.
+
+        GROMACS: .mpd,.top,.gro
+        OpenMM: (look at ExTasy workflows)
+        PLUMED-GROMACS: GROMACS + PLUMED input file
+        LAMMPS: .input file
+
+        Crucially, it provide a place for the information to be staged
+        for modifying input, such that a single simulation can be
+        loaded, and then variants of that simulation can be "cloned out", and modified.
+
+        For many programs, this may simply be putting the text of the
+        files into a container where they can be retrived and
+        modified.
+
     modify_input(simulation_input: SimulationInput, **kwargs) -> SimulationInput:
+
         A simulation input data source produced by *make_input* or *modify_input*
         may be modified in a pipe line through (implementation dependent) key word
         arguments to a *modify_input* command.
 
+        For most options, then a keyword argument can directly
+        parallel the keyword syntax used by the simulation programs
+        themselves: LAMMPS, PLUMED, GROMACS (in the .mdp) all use
+        keyword options.  It's not as clear for OpenMM, but we should
+        use the ExTASY implementations to guide us as to how modification is done there.  
+
+        One question is how the defintion of the chemical system
+        should changed using this approach. This is not set up very
+        well in the keyword format.  We could, for now, just replace
+        the file containing the molecular system definition entirely
+        using a modify_input.  This might be the best bet for now, but
+        we will want to think about what sorts of changes might be of
+        interest, if any in the future.
+
+        The issue is that in most cases, if the chemical system is
+        changing, it usually requires changes of many lines of
+        topologies, and so passing it in as a dictionary will likely
+        be a big pain. Likely easier to take care of it outside the
+        simulation for now. Eventually, if some parallel work on
+        simulation topology definitions in Open Force Field works,
+        then it may be possible to do this better.
+
+        Things like "training potential" that get attached in the
+        examples should probably be attached at this point in the
+        workflow, rather than at the simulate. For most workflows, it
+        could ONLY be added at this stage, since potentials are
+        defined in the input files.  For GMX_API driven simulations,
+        the actual attachment of the potentials to the md simulation
+        could happen in the simulate function, but it would be better
+        if they were loaded into the data structure in the
+        modify_input sage.
```

Comment:
This is possible, but I'm not sure it is desirably unless we can say that a "training potential" is something that can be implemented generically and reflected in other tools that operate on SimulationInput (like conversion to files or MDAnalysis objects). Right now, this is far from possible, and there is a useful distinction between an class of objects that don't have implementation-specific details and those that do, even if the representations (`scalems.wrappers.X.Simulation{,Input}`) are (currently) dependent on the tool set.

For instance, I would say that attaching the "training potential" modifies a Simulation object to produce another Simulation object, rather than modifying a SimulationInput object to produce another SimulationInput object. (Again, I might not have chosen the best names.) But I don't really see a reason to convert a SimulationInput object to a Simulation task and then modify the task in two separate steps. There has been user feedback on this exact issue, though. I think the feedback was that it was preferably to allow the Simulation to be specified in a single command, rather than created and then modified, but I'll try to double-check.

Reply:
> unless we can say that a "training potential" is something that can be implemented generically and reflected in other tools that operate on SimulationInput (like conversion to files or MDAnalysis objects)

Having a biasing potential is something most codes can handle, but generally it is defined by the input files.

Reply:
Yes, but the extension APIs for different packages are vastly different, and many researchers may only write code for one. If we want to accommodate tool-independent representation of attached add-on code, I suggest labeling a third reference type between SimulationInput and Simulation.